src/main/java/bitflow4j/steps/normalization/OnlineAutoMinMaxScaler.java

```diff
@@ -172,4 +177,57 @@ public String toString() {
         return "Online auto min-max scaler";
     }
 
+    protected static class ThresholdConceptChangeDetector implements ConceptChangeDetector {
+        private final double threshold;
+
+        public ThresholdConceptChangeDetector(double threshold) {
+            this.threshold = threshold;
+        }
+
+        @Override
+        public boolean isConceptChanged(double value) {
+            if (this.threshold > 0)
+                return value > threshold;
+            else
+                return false;
+        }
+    }
+
+    protected static class RobustThresholdConceptChangeDetector implements ConceptChangeDetector {
+        private static final int NUM_SAMPLES = 6;
+
+        private final double threshold;
+        private final int windowSize;
+        private List<Double> values;
+
+        public RobustThresholdConceptChangeDetector(double threshold) {
+            this(threshold, NUM_SAMPLES);
+        }
+
+        public RobustThresholdConceptChangeDetector(double threshold, int windowSize) {
+            this.threshold = threshold;
+            this.windowSize = windowSize;
+            this.values = new ArrayList<>();
+        }
+
+        @Override
+        public boolean isConceptChanged(double value) {
+            double referenceValue = value;
+            if (threshold > 0) {
+                values.add(value);
+                if (values.size() >= windowSize) {
+                    int index = Math.round((float) windowSize / 2);
+                    if (index >= 0 && index < values.size()) {
+                        referenceValue = values.get(index);
```

Comment:
I'm not sure I understand the logic behind this... Shouldn't this compute the average of the window or something? I I understand correctly, this should make everything the concept-changing robust to spikes, but this way the spike might randomly be the one selected comparsion value.

Reply:
Probably, the values list was meant to be sorted, before picking the value from the middle?... I don't want to make the fix myself before checking with @alek-thunder

Reply:
Very good point. Will fix this. My idea is to sort the window before accessing the middle element. Therefore sparse outliers (also in both directions for min max scaling) are ignored for determining the new min and max values that should be used for scaling.